src/cli/cli_dataset.cpp

```diff
@@ -445,14 +550,29 @@ otError Dataset::ProcessMeshLocalPrefix(uint8_t aArgsLength, char *aArgs[])
 otError Dataset::ProcessNetworkName(uint8_t aArgsLength, char *aArgs[])
 {
     otError error = OT_ERROR_NONE;
-    size_t  length;
 
-    VerifyOrExit(aArgsLength > 0, error = OT_ERROR_INVALID_ARGS);
-    VerifyOrExit((length = strlen(aArgs[0])) <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+    if (aArgsLength == 0)
+    {
+        if (sDataset.mComponents.mIsNetworkNamePresent)
+        {
+            mInterpreter.mServer->OutputFormat("%.*s\r\n", static_cast<uint16_t>(sizeof(sDataset.mNetworkName)),
+                                               sDataset.mNetworkName.m8);
+        }
+        else
+        {
+            mInterpreter.mServer->OutputFormat("none\r\n");
```

Comment:
How can we tell if there is no `Network Name` in dataset, or the `Network Name` is "none" ?

Reply:
I would prefer (slightly) not to output anything if the corresponding field is not present in the dataset (not just `networkname`, but all fields)
Thoughts? @jwhui